### PR TITLE
Migrate from spotbugs to errorprone

### PR DESCRIPTION
**Pull request by anuraaga**
Consistent with SDK, prints errors to console instead of file, and easier to SuppressWarnings

### CODE REVIEW

File: instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/TracedDelegatingConsumer.java [hunk 1]
```diff
@@ -62,35 +62,18 @@ public void handleRecoverOk(String consumerTag) {
...
-    Scope scope = null;
-    try {
-      context = tracer().startDeliverySpan(queue, envelope, properties, body);
-      scope = context.makeCurrent();
+    Context context = tracer().startDeliverySpan(queue, envelope, properties, body);
+    Scope scope = context.makeCurrent();
 
-    } catch (Exception e) {
-      log.debug("Instrumentation error in tracing consumer", e);
+    try {
```
Comment (mateuszrzeszutek) [inline instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/TracedDelegatingConsumer.java:68]: Nit: we can use try-with-resources here

File: instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/TracedDelegatingConsumer.java [hunk 2]
```diff
@@ -62,35 +62,18 @@ public void handleRecoverOk(String consumerTag) {
...
+      // Call delegate.
+      delegate.handleDelivery(consumerTag, envelope, properties, body);
+      tracer().end(context);
+    } catch (Throwable throwable) {
+      tracer().endExceptionally(context, throwable);
+      throw throwable;
     } finally {
-      // TODO this is very unusual code structure for this repo
-      // We have to review it
```
Comment (mateuszrzeszutek) [inline instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/TracedDelegatingConsumer.java:75]: 👍

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]
> Consistent with SDK

❤️ 

And great clean-ups 👍

File: instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/TracedDelegatingConsumer.java (hunk 1 above)
Comment (trask) [reply instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/TracedDelegatingConsumer.java]: done (so I can merge this before #3118)

**Review by BigGuy573 — COMMENTED** [review]
Tremendous Gain!  👍

**Pull request merged**